src/main/java/net/dv8tion/jda/internal/handle/GuildRoleUpdateHandler.java

```diff
@@ -88,6 +90,15 @@ protected Long handleInternally(DataObject content)
                             getJDA(), responseNumber,
                             role, oldColor));
         }
+        if (!Objects.equals(roleColors, RoleImpl.RoleColorsImpl.EMPTY))
```

Comment:
This will fire an event for every role update, this should compare against the old role colors.